tests/test_bytes_subclass_support.py

```diff
@@ -15,10 +19,10 @@ def test_handles_bytes_subclasses(self) -> None:
         """
 
         class CustomBytes(bytes):
-            def __new__(cls, data: str):
-                data_bytes = base64url_to_bytes(data)
-                self = bytes.__new__(cls, memoryview(data_bytes).tobytes())
-                return self
+            def __new__(cls, data: Union[str, bytes]):
+                return bytes.__new__(
+                    cls, data.encode("ascii") if isinstance(data, str) else data
+                )
```

Comment:
What prompted this rewrite? It doesn't obviously seem driven by changes in Pydantic v2 🤔